pareto/strategic_water_management/strategic_produced_water_optimization.py

```diff
@@ -1800,6 +1817,15 @@ def create_model(df_sets, df_parameters, default={}):
         units=model.model_units["volume_time"],
         doc="Big-M flow parameter [volume/time]",
     )
+
+    model.p_M_Concentration = Param(
+        default=100,
+        units=model.model_units["concentration"],
+        doc="Big-M concentration parameter [concentration]",
+    )
+
+    model.p_M_Flow_Conc = model.p_M_Flow * model.p_M_Concentration
```

Comment:
I'm not sure this is a good way to define this big-M parameter. If we need a big-M parameter with units of flow * concentration, I would use another call to `Param()` to crate it.

Reply:
@tarnold17 I agree that the big M for the bilinear term of F*Q can be easily defined using only one parameter. However, the value of big M required for this term is defined with the units used. Because mass is conserved, mass balances in water treatment systems are commonly written as mass fraction times mass flow. Mass fraction is always between 0 and 1, which makes estimating the magnitude of the big M for mass fraction and flow times concentration intuitive. However, the current formulation of PARETO employs volume balances rather than mass balances, with concentration and flow units of kg/l and bbl/week. In adjusting the big M for the bilinear term F*Q, the current units are not intuitive. As a result, using a separate big M for concentration (kg/l) makes it easier for anyone working with PARETO to change it with any unit change. Furthermore, defining a big M for concentration may become necessary when setting upper bounds on concentration variables in future MINLP formulations with quality in bilinear terms.

Reply:
So, `model.p_M_Flow` and `model.p_M_Concentration` are both Pyomo `Param`s - specifically, the have the type `pyomo.core.base.param.ScalarParam`. However, when you multiply these variables together, the result that you get is no longer a `Param`, it's an expression. Specifically, it has type `pyomo.core.expr.numeric_expr.NPV_ProductExpression`. This seems like bad practice to me. which is why I suggest using a call to `Param()` to create `model.p_M_Flow_Conc`.

Reply:
@tarnold17 Thank you for raising a valid point about the efficiency of using an expression for the model.p_M_Flow_Conc parameter. To address your concern and avoid any confusion with the bilinear term unit, I have defined model.p_M_Flow_Conc as a separate parameter that is initialized using model.p_M_Flow and model.p_M_Concentration. This approach also allows us to keep model.p_M_Concentration as the big-M parameter for concentration and use it in our MINLP.